src/System.Drawing.Common/src/System/Drawing/Brushes.cs

```diff
@@ -151,2543 +147,178 @@ public sealed class Brushes
-                if (navajoWhite == null)
-                {
-                    navajoWhite = new SolidBrush(Color.NavajoWhite);
-                    SafeNativeMethods.Gdip.ThreadData[s_navajoWhiteKey] = navajoWhite;
-                }
-                return navajoWhite;
-            }
-        }
-
-        /// <include file='doc\Brushes.uex' path='docs/doc[@for="Brushes.Navy"]/*' />
-        /// <devdoc>
-        ///    A brush of the given color.
-        /// </devdoc>
-        public static Brush Navy
-        {
-            get
-            {
-                Brush navy = (Brush)SafeNativeMethods.Gdip.ThreadData[s_navyKey];
-                if (navy == null)
-                {
-                    navy = new SolidBrush(Color.Navy);
-                    SafeNativeMethods.Gdip.ThreadData[s_navyKey] = navy;
-                }
-                return navy;
-            }
-        }
-
-        /// <include file='doc\Brushes.uex' path='docs/doc[@for="Brushes.OldLace"]/*' />
-        /// <devdoc>
-        ///    A brush of the given color.
-        /// </devdoc>
-        public static Brush OldLace
-        {
-            get
-            {
-                Brush oldLace = (Brush)SafeNativeMethods.Gdip.ThreadData[s_oldLaceKey];
-                if (oldLace == null)
-                {
-                    oldLace = new SolidBrush(Color.OldLace);
-                    SafeNativeMethods.Gdip.ThreadData[s_oldLaceKey] = oldLace;
-                }
-                return oldLace;
-            }
-        }
-
-        /// <include file='doc\Brushes.uex' path='docs/doc[@for="Brushes.Olive"]/*' />
-        /// <devdoc>
-        ///    A brush of the given color.
-        /// </devdoc>
-        public static Brush Olive
-        {
-            get
-            {
-                Brush olive = (Brush)SafeNativeMethods.Gdip.ThreadData[s_oliveKey];
-                if (olive == null)
-                {
-                    olive = new SolidBrush(Color.Olive);
-                    SafeNativeMethods.Gdip.ThreadData[s_oliveKey] = olive;
-                }
-                return olive;
-            }
-        }
-
-        /// <include file='doc\Brushes.uex' path='docs/doc[@for="Brushes.OliveDrab"]/*' />
-        /// <devdoc>
-        ///    A brush of the given color.
-        /// </devdoc>
-        public static Brush OliveDrab
-        {
-            get
-            {
-                Brush oliveDrab = (Brush)SafeNativeMethods.Gdip.ThreadData[s_oliveDrabKey];
-                if (oliveDrab == null)
-                {
-                    oliveDrab = new SolidBrush(Color.OliveDrab);
-                    SafeNativeMethods.Gdip.ThreadData[s_oliveDrabKey] = oliveDrab;
-                }
-                return oliveDrab;
-            }
-        }
-
-        /// <include file='doc\Brushes.uex' path='docs/doc[@for="Brushes.Orange"]/*' />
-        /// <devdoc>
-        ///    A brush of the gi
```

Comment:
It could be a separate change, but we should consider changing the code to lazily initialize the s_ fields (the lazy initialization could be done inside GetBrush by changing the first arg to be a ref).  As it stands now, the code will create ~140 objects when any of the colors is first accessed, and what are the chances an app is actually going to use all or even most of those.